src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnosticsForSpan.cs

```diff
@@ -106,15 +106,24 @@ private class LatestDiagnosticsForSpanGetter
                 _includeSuppressedDiagnostics = includeSuppressedDiagnostics;
             }
 
-            public async Task<bool> TryGetAsync(List<DiagnosticData> list, CancellationToken cancellationToken)
+            public async Task<bool> TryGetAsync(List<DiagnosticData> list, Func<string, IDisposable> addOperationScopeOpt, CancellationToken cancellationToken)
             {
                 try
                 {
                     var containsFullResult = true;
                     foreach (var stateSet in _stateSets)
                     {
-                        using (RoslynEventSource.LogInformationalBlock(FunctionId.DiagnosticAnalyzerService_GetDiagnosticsForSpanAsync, stateSet.Analyzer, cancellationToken))
+                        IDisposable disposableScope = null;
+                        RoslynEventSource.LogBlock? logBlock = null;
+                        try
                         {
+                            if (addOperationScopeOpt != null)
+                            {
+                                var analyzerTypeName = this._owner.HostAnalyzerManager.GetAnalyzerTypeName(stateSet.Analyzer);
+                                disposableScope = addOperationScopeOpt.Invoke(analyzerTypeName);
```

Comment:
why nto actually just use 'usings' for these guys?

Reply:
question still applies.

Reply:
We want to execute the telemetry logging and messaging only on Ctrl + Dot code path, so it needs to be executed conditionally.